backend/api/Services/MissionDefinitionService.cs

```diff
@@ -0,0 +1,224 @@
+            return _context.MissionDefinitions
+                .Include(mission => mission.Area)
+                .ThenInclude(robot => robot.Deck)
+                .ThenInclude(robot => robot.Installation)
+                .ThenInclude(robot => robot.Asset)
+                .Include(mission => mission.Source)
+                .Include(mission => mission.LastRun)
+                .ThenInclude(planTask => planTask == null ? null : planTask.StartTime);
+        }
+
+        public async Task<MissionDefinition?> ReadById(string id)
+        {
+            return await GetMissionsWithSubModels()
+                .FirstOrDefaultAsync(mission => mission.Id.Equals(id));
+        }
+
+        public async Task<PagedList<MissionDefinition>> ReadAll(MissionDefinitionQueryStringParameters parameters)
+        {
+            var query = GetMissionsWithSubModels();
+            var filter = ConstructFilter(parameters);
+
+            query = query.Where(filter);
+
+            SearchByName(ref query, parameters.NameSearch);
+
+            ApplySort(ref query, parameters.OrderBy);
+
+            return await PagedList<MissionDefinition>.ToPagedListAsync(
+                query,
+                parameters.PageNumber,
+                parameters.PageSize
+            );
+        }
+
+        public async Task<MissionDefinition> Update(MissionDefinition mission)
+        {
+            var entry = _context.Update(mission);
+            await _context.SaveChangesAsync();
+            return entry.Entity;
+        }
+
+        public async Task<MissionDefinition?> Delete(string id)
+        {
+            // We do not delete the source here as more than one mission definition may be using it
+            var mission = await ReadById(id);
+            if (mission is null)
+            {
+                return null;
+            }
+
+            _context.MissionDefinitions.Remove(mission);
+            await _context.SaveChangesAsync();
+
+            return mission;
+        }
+
+        private static void SearchByName(ref IQueryable<MissionDefinition> missions, string? name)
+        {
+            if (!missions.Any() || string.IsNullOrWhiteSpace(name))
+                return;
+
+            missions = missions.Where(
+                mission =>
+                    mission.Name != null && mission.Name.ToLower().Contains(name.Trim().ToLower())
+            );
+        }
+
+        /// <summary>
+        /// Filters by <see cref="MissionDefinitionQueryStringParameters.AssetCode"/> and <see cref="MissionDefinitionQueryStringParameters.Status"/>
+        ///
+        /// <para>Uses LINQ Expression trees (see <seealso href="https://docs.microsoft.com/en-us/dotnet/csharp/expression-trees"/>)</para>
+        /// </summary>
+        /// <param name="parameters"> The variable containing the filter params </param>
+        private static Expression<Func<MissionDefinition, bool>> ConstructFilter(
+            MissionDefinitionQueryStringParameters parameters
+        )
```

Comment:
Can this function be refactored? See very similar code in `MissionRunService`

Reply:
Since the MissionDefinitionService filter is so much simpler than the MissionRun one, I think for now it is easier and more readable to keep them separate, but that is just my subjective opinion.